test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -238,24 +247,50 @@ func main() {
 	for iteration := 0; iteration != *upTo; iteration++ {
 		responses := make(map[string]int, *podsPerNode*len(nodes.Items))
 		start := time.Now()
-		for q := 0; q < queries; q++ {
-			t := time.Now()
-			hostname, err := c.Get().
-				Namespace(ns).
-				Prefix("proxy").
-				Resource("services").
-				Name("serve-hostnames").
-				DoRaw()
-			glog.V(4).Infof("Proxy call in namespace %s took %v", ns, time.Since(t))
-			if err != nil {
-				glog.Infof("Call failed during iteration %d query %d : %v", iteration, q, err)
+		batch := 0
+		for batch < queries {
+			var send int
+			if batch+batchSize >= queries {
+				send = queries % batchSize
 			} else {
-				responses[string(hostname)]++
+				send = batchSize
 			}
-
+			glog.V(3).Infof("Spawing a batch of %d requests", send)
+			responseChan := make(chan string, send)
+			go func(toSend int) {
```

Comment:
From what I can tell, this doesn't actually issue the 100 requests in parallel. It spins off a go routine to execute 100 sequential requests, while this go routine waits for them to finish. Am I missing something here?